docs/src/main/sphinx/security/jwt.rst

```diff
@@ -0,0 +1,144 @@
+keys are the counterpart of the private keys embedded in JSON web tokens.
+
+JWT authentication configuration
+--------------------------------
+
+Enable JWT authentication by setting the :doc:`JWT authentication type
+<authentication-types>` in :ref:`etc/config.properties <config_properties>`, and
+specifying a URL or path to a key file:
+
+.. code-block:: properties
+
+    http-server.authentication.type=JWT
+    http-server.authentication.jwt.key-file=https://cluster.example.net/.well-known/jwks.json
+
+JWT authentication is typically used in addition to other authentication
+methods:
+
+.. code-block:: properties
+
+    http-server.authentication.type=PASSWORD,JWT
+    http-server.authentication.jwt.key-file=https://cluster.example.net/.well-known/jwks.json
+
+The following configuration properties are available:
+
+.. list-table:: Configuration properties for JWT authentication
+   :widths: 50 50
+   :header-rows: 1
+
+   * - Property
+     - Description
+   * - ``http-server.authentication.jwt.key-file``
+     - Required. Specifies either the URL to a JWKS service or the path to a
+       PEM or HMAC file, as described below this table.
+   * - ``http-server.authentication.jwt.required-issuer``
+     - Specifies a string that must match the value of the JWT's
+       issuer (``iss``) field in order to consider this JWT valid.
+       The ``iss`` field in the JWT identifies the principal that issued the
+       JWT.
+   * - ``http-server.authentication.jwt.required-audience``
+     - Specifies a string that must match the value of the JWT's
+       Audience (``aud``) field in order to consider this JWT valid.
+       The ``aud`` field in the JWT identifies the recipients that the
+       JWT is intended for.
+   * - ``http-server.authentication.jwt.principal-field``
+     - String to identify the field in the JWT that identifies the
+       subject of the JWT. The default value is ``sub``. This field is used to
+       create the Trino principal.
+   * - ``http-server.authentication.jwt.user-mapping.pattern``
+     - A regular expression pattern to :doc:`map all user names
+       </security/user-mapping>` for this authentication system to the format
+       expected by the Trino server.
+   * - ``http-server.authentication.jwt.user-mapping.file``
+     - The path to a JSON file that contains a set of
+       :doc:`user mapping rules </security/user-mapping>` for this
+       authentication system.
+
+Use the ``http-server.authentication.jwt.key-file`` property to specify
+either:
+
+-  The URL to a JWKS endpoint service, where the URL begins with ``https://``.
+   The JWKS service must be reachable from the coordinator. If the coordinator's
+   local network locks down port access for security, then administrators must
+   either open a port to the JWKS, or look in the server's
+   ``/.well-known/jwks.json`` file. `RFC 5785
+   <https://tools.ietf.org/html/rfc5785>`_ describes the ``.well-known``
+   mechanism.
```

Comment:
I find this sentence confusing.  Maybe just simplify to:
```
If the coordinator is running in a secured or firewalled network, the administrator **may** have to open access to the JWKS server host.

Reply:
Do you mean: don't mention the .well-known part at all?